_drafts/bedrock-code-to-production.md

```diff
@@ -0,0 +1,231 @@
+
+## GitHub pull request
+
+Once our page is coded and we're ready for the next step, we can open a [pull request on GitHub](https://help.github.com/articles/creating-a-pull-request/). Once this is done the following things typically happen:
+
+  - Our continuous integration service, [CircleCI](https://circleci.com/), runs a series of automated checks on the pull request. This runs unit tests as well as checks the code for both syntax and style errors. Running these automated checks saves us significant time during code review by picking out routine errors.
+  - The pull request must then be code reviewed by another human. Every code change gets reviewed by at least one other developer before merging.
+  - If the change requires localization, strings are extracted from the branch and checked by our l10n team. If they look good, they are sent out to our volunteers for translation. Once this has begun, it is **very bad** to make a breaking change to a piece of copy.
+  - Once the pull request is approved by a reviewer, it can then be merged into the master branch and automatically deployed to our dev environment.
+
+## Other useful things to know about
+
+### Demo servers
+
+Sometimes we'd like to have a URL we can give people to demonstrate changes that aren't yet live on the site and that
+aren't yet ready to be merged into our master branch. We've enabled this via our [Jenkins CI](https://jenkins.io/) server
+and Github. A developer that would like to demo their changes has only to follow these steps:
+
+1. Push their git branch to the [primary repository](https://github.com/mozilla/bedrock/) using a special branch name.
+  - Branch name must start with `demo__` and then be any letters, numbers, and dashes (e.g. demo__the-dude)
+2. Profit
+
+That's really it. Jenkins will build the demo and if it is successful a notification in our IRC channel will inform
+the developer of their new demo's URL. Feel free to check out
+[how that works](https://github.com/mozilla/bedrock/blob/139e8e589b2ebe391cb159068152408e9f090687/Jenkinsfile#L163-L203) if you're curious.
+
+### A/B testing
+
+For A/B testing bedrock has two options available, [Traffic Cop](http://bedrock.readthedocs.io/en/latest/mozilla-traffic-cop.html) and [Optimizely](https://www.optimizely.com/). [Jon](https://github.com/jpetto/) wrote a great blog post on the pros and cons of using each solution so we won't repeat them here. You should [go read it]({{ site.baseurl }}/2017/01/16/traffic-cop/).
+
+### Analytics
+
+Analytics on bedrock are provided using [Google Tag Manager](https://www.google.com/analytics/tag-manager/) (GTM). Most of bedrocks shared components and pages are pre-built for analytics using common [event handlers and data attributes](http://bedrock.readthedocs.io/en/latest/analytics.html#gtm-listeners-data-attributes). We can also create and fire [custom events](http://bedrock.readthedocs.io/en/latest/analytics.html#datalayer-push-example) as required.
```

Comment:
Small nit - missing apostrophe. Should be `bedrock's shared components` (not `bedrocks`).